src/Symfony/Component/HttpFoundation/InputBag.php

```diff
@@ -0,0 +1,96 @@
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpFoundation;
+
+use Symfony\Component\HttpFoundation\Exception\BadRequestException;
+
+/**
+ * InputBag is a container for user input values such as $_GET, $_POST, $_REQUEST, and $_COOKIE.
+ *
+ * @author Saif Eddin Gmati <saif.gmati@symfony.com>
+ */
+final class InputBag extends ParameterBag
+{
+    /**
+     * Returns a string input value by name.
+     *
+     * @param string|null $default The default value if the input key does not exist
+     *
+     * @return string|null
+     */
+    public function get(string $key, $default = null)
+    {
+        if (null !== $default && !is_scalar($default) && !\is_callable([$default, '__toString'])) {
+            @trigger_error(sprintf('Passing a non-string value as 2nd argument to "%s::%s()" is deprecated since Symfony 5.1, pass a string or null instead.', static::class, __METHOD__), E_USER_DEPRECATED);
+        }
+
+        $value = parent::get($key, $this);
+
+        if (null !== $value && $this !== $value && !is_scalar($value) && !\is_callable([$value, '__toString'])) {
+            @trigger_error(sprintf('Retrieving a non-string value from "%s::%s()" is deprecated since Symfony 5.1, and will throw a "%s" exception in Symfony 6.0, use "%1$s::all()" instead.', static::class, __METHOD__, BadRequestException::class), E_USER_DEPRECATED);
+        }
+
+        return $this === $value ? $default : $value;
+    }
+
+    /**
+     * Returns the inputs.
+     *
+     * @param string|null $key The name of the input to return or null to get them all
+     */
+    public function all(string $key = null): array
+    {
+        if (null !== $key) {
+            $value = $this->parameters[$key] ?? [];
+            if (!\is_array($value)) {
+                throw new Exception\BadRequestException(sprintf('Unexpected value for %s input.', $key));
+            }
+
+            return $value;
+        }
+
+        return $this->parameters;
+    }
+
+    /**
+     * Replaces the current input values by a new set.
+     */
+    public function replace(array $inputs = [])
+    {
+        $this->parameters = [];
+        $this->add($inputs);
+    }
+
+    /**
+     * Adds input values.
+     */
+    public function add(array $inputs = [])
+    {
+        foreach ($inputs as $input => $value) {
+            $this->set($input, $value);
+        }
+    }
+
+    /**
+     * Sets an input by name.
+     *
+     * @param string|array $value The value
+     */
+    public function set(string $key, $value)
+    {
+        if (!is_scalar($value) && !\is_callable([$value, '__toString']) && !\is_array($value)) {
+            @trigger_error(sprintf('Passing "%s" as a 2nd Argument to "%s::%s()" is deprecated since Symfony 5.1, pass a string or an array instead.', \is_object($value) ? \get_class($value) : \gettype($value), static::class, __METHOD__), E_USER_DEPRECATED);
```

Comment:
`static::class` should be replaced by `__CLASS__`: only this method is concerned, we don't know about child classes (same in other places in the PR)